text/0000-adding-pyo3-like-attribute-macros.md

```diff
@@ -0,0 +1,75 @@
+- Feature Name: adding_pyo3_like_attribute_macros_for_classes_and_methods
+- Start Date: 2021-06-02
+- RFC PR: (leave this empty)
+- Neon Issue: (leave this empty)
+
+# Summary
+[summary]: #summary
+
+This RFC proposes adding [PyO3-like](https://pyo3.rs/v0.13.2/class.html) attribute-macros to define JavaScript classes as Rust structs. 
+
+# Motivation
+[motivation]: #motivation
+
+At the moment of migration to `napi`, there is no idiomatic way of writing JavaScript classes in pure Rust. This proposal adds the attribute macros to help users to write classes and methods easily. 
+
+# Guide-level explanation
+[guide-level-explanation]: #guide-level-explanation
+
+I will describe some potential use cases here:
+```rust
+#[neon::class]
+struct Person {
+    name: string,
+    age: u8,
+}
+#[neon::methods]
```

Comment:
I think these should be namespaced to help prevent conflicts with other macros. This could be named `neon::class_methods` or we could have everything in a `neon::class` module.

```rust
#[neon::class::data]
#[neon::class::methods]
```

Reply:
Currently neon only has one proc/attribute macro, right? `main`

Reply:
That's correct.

Reply:
I actually prefer the simplicity of `#[neon::methods], #[neon::classes]` if in a foreseeable future there won't be any namespace conflicts.